File: VB/Core/CommonVB.Wpf/Helpers/DispatcherHelper.vb
Imports System.Windows
Imports System.Windows.Threading

Public Module DispatcherHelper

    Public Sub Execute(action As Action)

        If Application.Current Is Nothing OrElse Application.Current.Dispatcher Is Nothing Then
            Return
        End If

        ' Marshall to Main Thread
        Application.Current.Dispatcher.BeginInvoke(DispatcherPriority.Background, action)

    End Sub

End Module

File: CSharp/Core/Common.Wpf/Helpers/DispatcherHelper.cs
namespace System.Windows.Threading;

public static class DispatcherHelper
{
    public static void Execute(Action action)
    {
        if (Application.Current is null || Application.Current.Dispatcher is null)
            // We are already on the Main Thread
            return;

        // Marshall to Main Thread
        Application.Current.Dispatcher.BeginInvoke( DispatcherPriority.Background, action);
    }
}